app/stores/WalletsStore.js

```diff
@@ -194,6 +220,14 @@ export default class WalletsStore extends Store {
       if (!hasActiveWallet && hasAnyWalletsLoaded) this.active = this.all[0];
       if (this.active) this.goToWalletRoute(this.active.id);
     }
-  }
+  };
+
+  _patchWalletRequestWithWallet = async (wallet) => {
+    await this.walletsRequest.patch(result => {
+      const foundWallet = _.find(result, { id: wallet.id });
+      if (foundWallet) result.pop(foundWallet);
+      result.push(wallet);
+    });
+  };
```

Comment:
@DominikGuzei: We had a race condition here, polling wallets sometimes got the new wallet before we patched the request. Please take a look at this and merge if everything is ok.